src/array/polygon/array.rs

```diff
@@ -473,6 +473,17 @@ impl TryFrom<PolygonArray<i64>> for PolygonArray<i32> {
     }
 }
 
+impl<O: OffsetSizeTrait> From<RectArray> for PolygonArray<O> {
+    fn from(value: RectArray) -> Self {
+        let output_geoms: Vec<Option<geo::Polygon>> = value
+            .iter_geo()
+            .map(|maybe_g| maybe_g.map(|geom| geom.to_polygon()))
+            .collect();
+        let mut_arr: MutablePolygonArray<O> = output_geoms.into();
+        mut_arr.into()
+    }
```

Comment:
We should use the code from https://github.com/geoarrow/geoarrow-rs/blob/57c0db9870777aa39334bc844b3fcfda5a4b855a/src/algorithm/geo/minimum_rotated_rect.rs#L37-L53 to minimize allocations.

In particular, we know that [`Rect::to_polygon`](https://docs.rs/geo/latest/geo/geometry/struct.Rect.html#method.to_polygon) will always generate a simple polygon with exactly 5 coordinates, so we can reserve exactly the sizes of buffers we'll need

Reply:
Done in https://github.com/geoarrow/geoarrow-rs/pull/244